buildpack.toml

```diff
@@ -12,6 +12,18 @@ api = "0.8"
 [metadata]
   include-files = ["bin/run", "bin/build", "bin/detect", "buildpack.toml"]
   pre-package = "./scripts/build.sh"
+  [metadata.default-versions]
+    vsdbg = "17-2-10518-1"
+
+[[metadata.dependencies]]
+id = "vsdbg"
+name = "Visual Studio Debugger"
+sha256 = "dbd5c399214ccdc88cf19eca550b3351eb38e4534b55dafcafed4197ef5fd253"
+source = "https://vsdebugger.azureedge.net/vsdbg-17-2-10518-1/vsdbg-linux-x64.tar.gz"
+source_sha26 = "dbd5c399214ccdc88cf19eca550b3351eb38e4534b55dafcafed4197ef5fd25"
+stacks = ["io.buildpacks.stacks.bionic","io.buildpacks.stacks.jammy"]
+uri = "https://vsdebugger.azureedge.net/vsdbg-17-2-10518-1/vsdbg-linux-x64.tar.gz"
+version = "17-2-10518-1"
```

Comment:
It'd be great if we could have a PURL and/or CPE for this dependency. Is that plausible to find/generate? I know this dependency is a bit weird...

Reply:
Agreed! You can go to the NIST CPE lookup page and see what you can find for it

Reply:
https://nvd.nist.gov/products/cpe/search

Reply:
I wasn't able to find a CPE for it when I looked (I tried `vsdbg` as well as `visual studio debugger`), which is why I went for the Generate() call instead of GenerateFromDependency. I figured that way the SBoM would at least have a record of the name of the tool.

Reply:
Ah, I didn't realize that `GenerateFromDependency()` would fail if there are no CPEs for the dependency. That's a shame. From what I can see, the current SBOM output is:
```
{
 "artifacts": [],
 "artifactRelationships": [],
 "source": {
  "type": "directory",
  "target": "/layers/paketo-buildpacks_vsdbg/vsdbg"
 },
 "distro": {},
 "descriptor": {
  "name": "",
  "version": ""
 },
 "schema": {
  "version": "3.0.1",
  "url": "https://raw.githubusercontent.com/anchore/syft/main/schema/json/schema-3.0.1.json"
 }
}
```

I do feel a bit worried that the SBOM is so minimal - given that there's no CPE or PURL, it's really hard for a user (or auditor) to know what the heck this thing _is_.  Maybe we can make `GenerateFromDependency` a bit more flexible when CPEs/PURLs are missing?

Reply:
I would agree that this should not fail when CPE or PURL are missing. The primary use case for a CPE or PURL is to identify the package in case it needs to be patched for a CVE. If Microsoft doesn't provide a CPE for `vsdbg`, then there isn't much that we can do.

Reply:
I think it'd be great to at least get the vsdbg version and dependency name from the `buildpack.toml` into the SBOM. So, I think the next move is to make `GenerateFromDependency()` work for dependencies that are missing CPEs and/or PURLs.

Reply:
We have in the past tried to make our own PURLs based on the name of the dependency. If this is something that we should not be doing then we may want to evaluate when going over our dependencies in the future.

Reply:
@ForestEckhardt If we can create our own PURLs then I'm happy to do at least that. Genuine ignorance here, but what would the utility of that be? What/who would consume that PURL in a way that is useful?

Reply:
After getting some feedback from stakeholders about CPEs, I think our best move is to create a "best-effort" CPE for this dependency that includes the component type (`a` for software application), vendor name (microsoft) and the program name. I think that'd be
```
cpe:2.3:a:microsoft:vsdbg:*:*:*:*:*:*:*:*
```
We should use the new `cpes` buildpack dependency field in lieu of the deprecated `cpe` field. See https://github.com/paketo-buildpacks/packit/pull/377 for details.